includes/class-gdpr.php

```diff
@@ -331,9 +331,11 @@ public static function similar_in_array( $needle, $haystack ) {
 	public static function save_user_consent_on_registration( $user_id ) {
 		GDPR_Audit_Log::log( $user_id, esc_html__( 'User registered to the site.', 'gdpr' ) );
 
+		// phpcs:disable WordPressVIPMinimum.Security.PHPFilterFunctions.RestrictedFilter
 		// phpcs:disable WordPressVIPMinimum.VIP.PHPFilterFunctions.RestrictedFilter
 		$user_consents = filter_input( INPUT_POST, 'user_consents', FILTER_DEFAULT, FILTER_REQUIRE_ARRAY );
 		// phpcs:enable WordPressVIPMinimum.VIP.PHPFilterFunctions.RestrictedFilter
+		// phpcs:enable WordPressVIPMinimum.Security.PHPFilterFunctions.RestrictedFilter
```

Comment:
Can we change `FILTER_DEFAULT` to `FILTER_SANITIZE_STRING` instead of just disabling?

Reply:
We don't seem to use it, it always seems to be `null`. What's concerning me from simply removing it is I can't figure out how it is supposed to be used, and of course, there are no unit tests so figuring out the way it's supposed to work is a crapshoot. 

In short:

`GDPR_Audit_Log::log( $user_id, esc_html__( 'User registered to the site.', 'gdpr' ) );`

is used.

But as the `$user_contents` outcome is always(?) `null`, then the conditional is always `false`, so all we have here is a VIP warning with absolutely no benefit to production code.

Reply:
ok figured it out, it's for if we were using custom consents on registration. We aren't, I've removed the entire conditional.

Reply:
we aren't but if it's supported we should leave the plugin be fairly agnostic?

Reply:
ok, no problem